packages/caliper-ethereum/lib/ethereum.js

```diff
@@ -0,0 +1,213 @@
+    }
+
+    /**
+     * Release the given Ethereum context.
+     * @param {object} context The Ethereum context to release.
+     * @async
+     */
+    async releaseContext(context) {
+        // nothing to do
+    }
+
+    /**
+     * Invoke a smart contract.
+     * @param {Object} context Context object.
+     * @param {String} contractID Identity of the contract.
+     * @param {String} contractVer Version of the contract.
+     * @param {Array} invokeData Array of methods calls.
+     * @param {Number} timeout Request timeout, in seconds.
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async invokeSmartContract(context, contractID, contractVer, invokeData, timeout) {
+        let promises = [];
+        invokeData.forEach((item, index) => {
+            promises.push(this.sendTransaction(context, contractID, contractVer, item, timeout));
+        });
+        return Promise.all(promises);
+    }
+
+    /**
+     * Submit a transaction to the ethereum context.
+     * @param {Object} context Context object.
+     * @param {String} contractID Identity of the contract.
+     * @param {String} contractVer Version of the contract.
+     * @param {Object} methodCall Array of JSON containing methods data.
+     * @param {Number} timeout Request timeout, in seconds.
+     * @return {Promise<TxStatus>} Result and stats of the transaction invocation.
+     */
+    async sendTransaction(context, contractID, contractVer, methodCall, timeout) {
+        let status = new TxStatus();
+        try {
+            context.engine.submitCallback(1);
+            let receipt = null;
+            if (methodCall.args) {
+                receipt = await context.contracts[contractID].methods[methodCall.verb](...methodCall.args).send({from: context.fromAddress});
+            } else {
+                receipt = await context.contracts[contractID].methods[methodCall.verb]().send({from: context.fromAddress});
+            }
+            status.SetID(receipt.transactionHash);
+            status.SetResult(receipt);
+            status.SetVerification(true);
+            status.SetStatusSuccess();
+        } catch (err) {
+            status.SetStatusFail();
+            logger.error('Failed tx on ' + contractID + ' calling method ' + methodCall.verb);
+        }
+        return Promise.resolve(status);
+    }
+
+    /**
+     * Query the given smart contract according to the specified options.
+     * @param {object} context The Ethereum context returned by {getContext}.
+     * @param {string} contractID The name of the contract.
+     * @param {string} contractVer The version of the contract.
+     * @param {string} key The argument to pass to the smart contract query.
+     * @param {string} [fcn=query] The contract query function name.
+     * @return {Promise<object>} The promise for the result of the execution.
+     */
+    async queryState(context, contractID, contractVer, key, fcn = 'query') {
```

Comment:
This is a legacy API function. Implement (and also leave this for now) the `querySmartContract` function (see the interface/baseclass). It's really similar to `invokeSmartContract`, so if web doesn't differentiate between queries and invokes, then you can reuse the `sendTransaction` function (handle the TX id setting, if queries don't have Tx ids).